notifications/bridge.go

```diff
@@ -23,8 +24,8 @@ var _ Listener = &bridge{}
 
 // URLBuilder defines a subset of url builder to be used by the event listener.
 type URLBuilder interface {
-	BuildManifestURL(name, tag string) (string, error)
-	BuildBlobURL(name string, dgst digest.Digest) (string, error)
+	BuildManifestURL(name reference.Named, tag string) (string, error)
```

Comment:
I've implemented this change at https://github.com/aaronlehmann/distribution/tree/use-reference-package-v2. It depends on https://github.com/docker/distribution/pull/1268 and https://github.com/docker/distribution/pull/1271, so I'm going to close this PR for now and reopen it when those PRs are merged.

(Slight correction to the above: `BuildManifestURL` will take a single `reference.Named` that may include a digest or tag.)